x-pack/plugins/reporting/server/config/create_config.ts

```diff
@@ -43,10 +43,22 @@ export function createConfig$(
       }
       const { kibanaServer: reportingServer } = config;
       const serverInfo = core.http.getServerInfo();
-      // kibanaServer.hostname, default to server.host
-      const kibanaServerHostname = reportingServer.hostname
+      // set kibanaServer.hostname, default to server.host, don't allow "0.0.0.0" as it breaks in Windows
+      let kibanaServerHostname = reportingServer.hostname
         ? reportingServer.hostname
         : serverInfo.hostname;
+      if (kibanaServerHostname === '0.0.0.0') {
```

Comment:
Is there a way that this could ever evaluate to `true` if we are validating the config at startup?

Reply:
This will evaluate to `true` when kibana.yml has `server.host: 0.0.0.0` and the reporting settings are all using the defaults.

I just noticed the unit test does not test the proper failover logic. Thanks!! I fixed it here: e6c0d52aa6b